flocker/control/_model.py

```diff
@@ -1232,10 +1232,50 @@ def get_information_wipe(self):
         return NoWipe()
 
 
+def _generation_hash_value_factory(x):
+    """
+    Factory method to create a generation hash.
+
+    This is specifically designed to be able to turn a ``bytes`` string into a
+    tuple of ints. See ``GenerationHash`` for why this is required.
+
+    :param x: The input value to be turned into a proper hash value in a
+        ``GenerationHash``.
+
+    :returns: A tuple of ints.
+    """
+    if x is None:
+        return x
+    transform = lambda y: y
+    if isinstance(x, bytes):
+        transform = ord
+    return tuple(transform(i) for i in x)
+
+
+class GenerationHash(PClass):
+    """
+    Generation hash uniquely defines a configuration or state. A value of None
+    indicates that there is no configuration or state known.
+
+    Note that this is a tuple of ints rather than ``bytes`` because the flocker
+    ``wire_encode`` function assumes that ``bytes`` can be encoded as JSON
+    strings, which is not true of generation hashes.
```

Comment:
Yea, that'd be equivalent, and would serialize to something smaller as well, I guess then in make_generation_hash I could compute the base64 encoding and put it in this object. I don't think there is a strong argument to go for one over the other.

On 1 hand base64 serializes smaller, on the other hand python doesn't have a good type to verify that a value is a base64 string... (I guess I could write a pyrsistent invariant or something...)